src/main/java/io/vertx/core/net/TCPSSLOptions.java

```diff
@@ -405,6 +410,50 @@ public TCPSSLOptions addCrlPath(String crlPath) throws NullPointerException {
   }
 
   /**
+   * Add key options for SNI host.
+   *
+   * @param domain The domain name to match. May contain wildcards.
+   * @param keyCertOptions The key options to use when matching this host.
+   * @return a reference to this, so the API can be used fluently
+   */
+  public TCPSSLOptions addSNIKeyCertOptionsForDomain(String domain, KeyCertOptions keyCertOptions) throws NullPointerException {
```

Comment:
- rename `addSniKeyCertOption`
- annotate it with @ GenIgnore
- add new adder / getter `addSniKeyStoreOption/getSniKeyStoreOptions`, `addSniPfxKeyCertOption/getSniPfxKeyCertOptions`, `addSniPemKeyCertOption/getSniPemKeyCertOptions` , not annotated with @ GenIgnore - follow the pattern used by keyCertOptions / setKeyStoreOption / setPfxKeyCertOption / setPemKeyCertOption